src/SFA.DAS.LevyTransferMatching.Web/Controllers/OpportunitiesController.cs

```diff
@@ -2,24 +2,69 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using SFA.DAS.LevyTransferMatching.Web.Orchestrators;
+using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
+using System.Data;
+using SFA.DAS.Authorization.Mvc.Attributes;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Controllers
 {
-    [AllowAnonymous]
-    [Route("opportunities")]
     public class OpportunitiesController : Controller
     {
-        private readonly IOpportunitiesOrchestrator _searchFundingOrchestrator;
+        private readonly IOpportunitiesOrchestrator _opportunitiesOrchestrator;
 
         public OpportunitiesController(IOpportunitiesOrchestrator searchFundingOrchestrator)
         {
-            _searchFundingOrchestrator = searchFundingOrchestrator;
+            _opportunitiesOrchestrator = searchFundingOrchestrator;
         }
 
         public async Task<IActionResult> Index()
         {
-            var viewModel = await _searchFundingOrchestrator.GetIndexViewModel();
+            var viewModel = await _opportunitiesOrchestrator.GetIndexViewModel();
             return View(viewModel);
         }
+
+        [Route("opportunities/{encodedId}")]
+        public async Task<IActionResult> Detail(string encodedId)
+        {
+            var viewModel = await _opportunitiesOrchestrator.GetDetailViewModel(encodedId);
+
+            if (viewModel != null)
+            {
+                return View(viewModel);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost]
+        [Route("opportunities/{encodedId}")]
+        public IActionResult ConfirmOpportunitySelection(string encodedId, DetailPostRequest detailPostRequest)
```

Comment:
We could move the encoded Id into the PostRequest.

Also, the name of the post request should match the endpoint name.

Is this a post action from the Detail view? If so, it should match the corresponding Get, so be called `Detail` also.

Reply:
* Moved `EncodedPledgeId` into post request model
* Renamed `DetailPostRequest` to match endpoint name -> `OpportunitiesPostRequest`
  * Also renamed associated validator `DetailPostRequestValidator` -> `OpportunitiesPostRequestValidator`
  * Also renamed associated unit tests
* Action renamed `ConfirmOpportunitySelection` -> `Detail`